src/auth/signup.js

```diff
@@ -0,0 +1,164 @@
+const InputMessage = {
+  ERROR_EMPTY_EMAIL: "이메일을 입력해주세요.",
+  ERROR_INVALID_EMAIL_FORMAT: "올바른 이메일 주소가 아닙니다.",
+  ERROR_DUPLICATE_EMAIL: "이미 사용 중인 이메일입니다.",
+  ERROR_EMPTY_PASSWORD: "비밀번호를 입력해주세요.",
+  ERROR_INVALID_PASSWORD_FORMAT:
+    "비밀번호는 영문,숫자 조합 8자 이상 입력해 주세요.",
+  ERROR_DIFFER_PASSWORD_CONFIRM: "비밀번호가 일치하지 않아요",
+  ERROR_NO_MATCHING_EMAIL: "이메일을 확인해 주세요.",
+  ERROR_NO_MATCHING_PASSWORD: "비밀번호를 확인해 주세요.",
+};
+
+const getChildElementsAsNode = (query) =>
+  Array.from(document.querySelector(query)?.childNodes).filter(
+    (node) => node.nodeType === Node.ELEMENT_NODE
+  );
+
+const signupForm = document.querySelector(".signup");
+const [_, emailInputEl, emailMessageEl] = getChildElementsAsNode(".email");
+const [__, passwordInputEl, passwordMessageEl] =
+  getChildElementsAsNode(".password");
+const [___, passwordConfirmInputEl, passwordConfirmMessageEl] =
+  getChildElementsAsNode(".password-confirm");
+const passwordToggleButtonEls = document.querySelectorAll(".password-toggle");
+
+const showInputErrorMessage = (inputEl, message) => {
+  const messageEl = inputEl.parentElement.lastElementChild;
+  messageEl.textContent = message;
+  messageEl.dataset.state = "error";
+};
+
+const hideInputErrorMessage = (inputEl) => {
+  inputEl.parentElement.lastElementChild.dataset.state = "normal";
+};
+
+const validateEmail = (email) => {
+  if (email === "")
```

Comment:
아래처럼 falsy한 값을 걸러줘도 좋을거 같네요

```suggestion
  if (!email)
```